Vietnam War Vessels Mod v0.9.0 integration

Astro-739 opened the pull request:
YAMLs, icons, weapon LUAs and integration files for Vietnam War Vessels Mod v0.9.0

https://forum.dcs.world/topic/338387-tetets-vietnam-war-vessels/
https://github.com/tspindler-cms/tetet-vwv/releases

Aircraft:  A-1H Skyrainder, F-8E Crusader, RA-5C Vigilante, O-1E Bird Dog, HH-2D and SH-2F Seasprite, MiG-17F, MiG-21MF
Ships: CVA-31 Bon Homme Richard, USS Allen M. Sumner (DD-692), USS Fletcher FRAM II Destroyer, USS Laffey (DD-724), USS Maddox (DD-731), USS The Sullivans (DD-537), P 4 Torpedo Boat, Patrol Boat, River MkII

Units have been added to Factions: Vietnam 1970, USA 1970 Vietnam War, USA 1970

Astro-739 committed:
vietnamwarvessels first batch

Astro-739 committed:
Ship YAMLs

Astro-739 committed:
aircraft yamls initial version, need more work

Astro-739 committed:
initial helicopter yamls

Astro-739 committed:
update aircraft yamls

Astro-739 committed:
Merge branch 'dcs-retribution:dev' into vietnamwarvessels

Astro-739 committed:
Added DDs Fletcher and Sullivans

Astro-739 committed:
ship icons

Astro-739 committed:
aircraft banners and icons

Astro-739 committed:
no huts

Astro-739 committed:
update py files to VWV v0.9.0

Astro-739 committed:
update aircraft yamls, add vigilante

Astro-739 committed:
added 2 ships for VWV v0.9.0

Astro-739 committed:
mig-21mf yaml

Astro-739 committed:
icons and banners additional units v0.9.0

Astro-739 committed:
added VWV units to USA_1970 and Vietnam_1970 JSONs

Astro-739 committed:
Revert "added VWV units to USA_1970 and Vietnam_1970 JSONs"

This reverts commit ed0b28dc36c0d9c1a45a10689a3c419bd23ff258.

Astro-739 committed:
A-1H yaml update

Astro-739 committed:
mig-17 yaml update

Astro-739 committed:
update helicopter yamls

Astro-739 committed:
extension init

Astro-739 committed:
weapon injections

Astro-739 committed:
icon filenames _24 added

Astro-739 committed:
removed tasks 0 from yamls

Astro-739 committed:
hh2d yaml fix

Astro-739 committed:
added VWV v0.9.0 to factions USA and Vietnam

Astro-739 committed:
added max_range to aircraft yamls

Astro-739 committed:
housekeeping

Astro-739 committed:
Flyable to False - not available in mod version

Astro-739 committed:
minor edits

Astro-739 committed:
ignore test campaign

Astro-739 committed:
deleted tasks

Astro-739 committed:
weapon luas blue air

Astro-739 committed:
added task numbers from task.py

Astro-739 committed:
weapon luas red air

Astro-739 committed:
task id numbers in comment

Astro-739 committed:
switched weapon lua from aim-9J to aim-9D

Astro-739 commented:
Aircraft are flying and ships are sailing. But still needs some work on the weapon luas. Not all aircraft get a weapon payload, Strike and Anti-Ship do not get a payload. I matched the task ids, but something is not right there.
Also the integration of the CV needs to be done. I don't know what is needed here.

Astro-739 committed:
Merge branch 'dcs-retribution:dev' into vietnamwarvessels

Raffson commented on .gitignore:
```diff
@@ -25,3 +25,5 @@ env/
 /resources/plugins/pretense/pretense_output.lua
 
 *.psd
+resources/campaigns/ASTRO-TEST-exercise_vegas_nerve.miz
+resources/campaigns/ASTRO-TEST-exercise_vegas_nerve.yaml
```
No need for this, you can move your test-campaigns to your saved games directory:
<Drive>:\<path_to_saved_games_folder>\DCS\Retribution\Campaigns

Astro-739 replied:
Ok, will move the files.

Raffson commented on resources/customized_payloads/vwv_crusader.lua:
```diff
@@ -0,0 +1,210 @@
+	{
+		[1]=
+		{
+			["displayName"]="Retribution STRIKE",
```
In this case it should actually be `Retribution Strike` 😅

Raffson commented on resources/customized_payloads/vwv_crusader.lua:
```diff
@@ -0,0 +1,210 @@
+		[4]=
+		{
+			["displayName"]="Retribution ESCORT",
+			["name"]="Retribution ESCORT",
```
Here's it's `Retribution Escort` 😅

Raffson commented on resources/customized_payloads/vwv_crusader.lua:
```diff
@@ -0,0 +1,210 @@
+		[5]=
+		{
+			["displayName"]="Retribution FIGHTER SWEEP",
+			["name"]="Retribution FIGHTER SWEEP",
```
`Retribution Fighter sweep` --> Reference the following for the mission types: https://github.com/dcs-retribution/dcs-retribution/blob/f36526b5de2377752b2d7392c2b62f6dc863d201/game/ato/flighttype.py#L41-L63

There are a few legacy names such as `STRIKE`, `CAP`, `CAS`, etc. without the preceding `Retribution` text, which are usually upper-cased, but for all payload names that contain `Retribution` in the name, the flight-types' values should be referenced.

Astro-739 replied:
Ok, thanks for the tips, will update the names.

Raffson commented on resources/customized_payloads/vwv_hh2d.lua:
```diff
@@ -0,0 +1,20 @@
+	{
+		[1]=
+		{
+			["displayName"]="Retribution AIR_ASSAULT",
```
Similar issue, although it seems that there are no pylons, so I guess it's not a real issue...

Raffson commented on resources/factions/usa_1970.json:
```diff
@@ -57,19 +63,27 @@
+    "USS Fletcher FRAM II Destroyer",
+    "USS Laffey (DD-724)",
+    "USS Maddox (DD-731)",
+    "USS The Sullivans (DD-537)"
```
Missing CVA-31 here?

Astro-739 replied:
I put the CV under carrier_names. I do not fully understand the difference between naval_units and carrier_names. Should the CV be under both?

Raffson replied:
Carrier names are simply names that are assigned to the carrier, but you'd still need to mention the unit type under naval unit, i.e. using the name as defined in the carrier's yaml file. But there is indeed something that need to be added in the code. I believe all that's needed is to add the class in the following file around line 1410: https://github.com/dcs-retribution/dcs-retribution/blob/dev/game/theater/controlpoint.py

Now I might still be missing something, so let me know if you run into any trouble 😅

Astro-739 replied:
Ok, thanks. I'll dive into the controlpoint.py file.
I discussed this Mod integration with Stoner, he mentioned the CV needs something extra, and he would look into that. So I don't know what is needed. I will shoot him a message as well.

Raffson replied:
That would be me, and I was mainly referring to that code in `controlpoint.py` but as I said, I might've missed something so let me know if you experience any issues after adding CVA-31 to that list 😅

Raffson replied:
Btw does it have LINK4 capabilities? If so, you'll also need to add the carrier on line 624 in tgogenerator.py, line 693 in pretenseluagenerator.py & line 718 in pretensegenerator.py

Astro-739 replied:
Success! Added the CV where you pointed out, all looks good, planes taking off and landing on the CV.

Astro-739 replied:
What? You are Stoner? Haha, ok, sorry, didn't know that.

Astro-739 replied:
Negative on the Link 4, so that makes it easier.
I also see CVs listed in radar_db.py. Do we need to do something there?

Raffson replied:
We might need to add the CVA-31 (and other ships that have a radar) to `UNITS_WITH_RADAR` in radar_db.py, although I'm not 100% sure i's really needed 😅

Raffson replied:
Either way, I think we're pretty close at this point, and I'm also noticing that other carriers which I've added in the past were never added to that list, and I've just confirmed that this list isn't used anywhere in the code, so for the sake of completeness you _could_ add it in case someone starts using it in the future, but currently wouldn't make much of a difference 😅

Raffson commented on resources/factions/USA 1970 Vietnam War.json:
```diff
@@ -64,23 +70,32 @@
+    "USS Fletcher FRAM II Destroyer",
+    "USS Laffey (DD-724)",
+    "USS Maddox (DD-731)",
+    "USS The Sullivans (DD-537)"
```
Missing CVA-31?

Raffson reviewed (changes requested):
Looks good overall, I'd just revisit the tasking in the yaml files one more time and add more Armed Recon to the units, and perhaps add Anti-ship for the helicopters?

Astro-739 committed:
removed test campaigns

Astro-739 committed:
Merge branch 'dcs-retribution:dev' into vietnamwarvessels

Astro-739 committed:
Merge branch 'vietnamwarvessels' of https://github.com/Astro-739/dcs-retribution into vietnamwarvessels

Astro-739 committed:
update payload luas with payload names from flighttype.py

Astro-739 committed:
Changed AIM-9D to 9B, 9D does not work

Astro-739 committed:
removed air assault task for HH-2D

Astro-739 committed:
Cva_31 added to runway_is_operational()

Astro-739 committed:
CVA-31 added to naval_units in faction jsons

Astro-739 commented:
I am going to look into adding more Armed Recon, but currently getting errors in retribution around armed recon flights, runtime errors about the aircraft not being capable of CAS or any of the fall-back tasks. I am adding extra tasks, but no success yet. Could this be in the weapon lua?

Raffson commented:
> I am going to look into adding more Armed Recon, but currently getting errors in retribution around armed recon flights, runtime errors about the aircraft not being capable of CAS or any of the fall-back tasks. I am adding extra tasks, but no success yet. Could this be in the weapon lua?

It basically means that the aircraft in question is not capable of CAS, AFAC or Anti-ship Strike in DCS, and thus won't be compatible with the tasking that is generated by Retribution.

Astro-739 committed:
add strike and cas tasks to ra-5c

Raffson commented on resources/factions/USA 1970 Vietnam War.json:
```diff
@@ -76,7 +76,8 @@
     "DDG Arleigh Burke IIa",
-    "CV-59 Forrestal",
+    "CVA-31 Bon Homme Richard",
+    "CV-59 USS Forrestal",
```
Should be changed back to "CV-59 Forrestal" (unit names in a faction file always reference the name that's specified within the unit's yaml file)

Raffson commented:
`black` seems to fail because there's a formatting issue in `controlpoint.py`, recommend running `black game` from the root of the project 😅

Astro-739 committed:
correct typo

Astro-739 committed:
Added Armed Recon as task and payload to most a/c

Astro-739 commented:
> `black` seems to fail because there's a formatting issue in `controlpoint.py`, recommend running `black game` from the root of the project 😅 

Some format changes were auto generated by my editor, could not commit without them. I"ll look into it.

Astro-739 commented:
I have added the armed recon task to most aircraft and their payloads. 
I am still having issues with the Vigilante. It generates the error I mentioned for all missions. I gave it a long list of tasks, nothing works. Also gave it fake payloads, no go. I had it flying before the payload luas, now it doesn't want to anymore. It doesn't have any payload, so it is not useful anyway. I am thinking to remove it from the factions.

Raffson commented on pydcs_extensions/vietnamwarvessels/vietnamwarvessels.py:
```diff
@@ -0,0 +1,1266 @@
+
+    pylons: Set[int] = set()
+
+    tasks = [task.Reconnaissance]
```
This explains why you're having with CAS & Armed Recon. Does Reconnaissance allow for an "Engage targets in zone" task?

Astro-739 committed:
ignore pre-commit-config.yaml

Astro-739 committed:
pre-commit-config

Astro-739 committed:
black reformat controlpoint.py

Astro-739 reviewed (commented):
I have reformatted controlpoint.py with black, format should be back to what it was.
Turned out the changes in format I introduced were according to black v22.6. I updated to black v24.8 and now it should be right again.

Astro-739 reviewed (commented):
> This explains why you're having with CAS & Armed Recon. Does Reconnaissance allow for an "Engage targets in zone" task? 

I'll look into it.

Astro-739 committed:
Added tasks to Vigilante (next to Recon) containing attack subtasks, which allow it to be scheduled for missions

Astro-739 reviewed (commented):
> This explains why you're having with CAS & Armed Recon. Does Reconnaissance allow for an "Engage targets in zone" task? 

Right, Reconnaissance does not have any attack subtasks.
I have added Strike and CAS to the tasks in vietnamwarvessels.py, and now it works, it flies all missions. BUT it now also gets scheduled for CAS missions when no other airtplanes are available, while not having any weapons... not ideal. For now let's leave it like this, and I might update later. What do you think?

Astro-739 committed:
added ships to UNITS_WITH_RADAR

Astro-739 commented:
> I have reformatted controlpoint.py with black, format should be back to what it was. Turned out the changes in format I introduced were according to black v22.6. I updated to black v24.8 and now it should be right again.

I just noticed that the Dev branch still has black v22.6 in pre-commit-config.yaml. This is what caused the difference in formatting. The version that is used to check PRs is v24.10. 

I use VScode with the Black Formatter extension which currently has black v24.8. 
Can you update pre-commit-config.yaml in dev to (at least) v24.8?

Raffson commented on .gitignore:
```diff
@@ -25,3 +25,4 @@ env/
 /resources/plugins/pretense/pretense_output.lua
 
 *.psd
+.pre-commit-config.yaml
```
No need to ignore this file, I use a different approach, hence it probably stuck with an old version 👍

Astro-739 commented:
Ok, I think we have everything now. 
Or do you have any remaining issues somewhere?

Astro-739 committed:
remove pre-commit-config from gitignore

Raffson commented:
> Ok, I think we have everything now. 
> 
> Or do you have any remaining issues somewhere?

I don't think, so let's go ahead and merge 👍

Raffson commented:
> Ok, I think we have everything now. 
> 
> Or do you have any remaining issues somewhere?

I don't think, so let's go ahead and merge 👍

Raffson commented:
Seems like there's some merge conflicts, so once those are fixed we should be able to merge 👍

Astro-739 committed:
Merge branch 'dev' into vietnamwarvessels

Astro-739 committed:
added red aircraft to nva_1970 faction

Astro-739 commented:
Resolved all conflicts. The China Military Asset Pack had updates in some of the same files.
Also added red a/c to NVA 1970 faction, as I just discussed with Starfire.

Astro-739 committed:
fixed black's complaint (two empty lines, should be one)
